Sources/Basic/SyncronizedQueue.swift

```diff
@@ -0,0 +1,58 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2015 - 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import class Foundation.Condition
+
+/// This class can be used as a shared queue between multiple threads providing
+/// thread safe APIs. The elements are stored as optionals so nil can be used as
+/// a sentinel.
```

Comment:
I think the client should just create a `SynchronizedQueue<Int?>` if they want that behavior, is there any reason not to do that?